atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -863,6 +863,13 @@ public String toString() {
                         RowColumnRangeExtractor extractor = new RowColumnRangeExtractor(metricsManager);
                         extractor.extractResults(ImmutableList.of(row), results, startTs);
                         RowColumnRangeExtractor.RowColumnRangeResult decoded = extractor.getRowColumnRangeResult();
+                        if (decoded.getResults().isEmpty()) {
+                            // All returned results were at ts > startTs
+                            // Start the next page from the last column seen to ensure forward progress
+                            byte[] lastColumn = extractor.getLastColumnForRow(row).get();
+                            return SimpleTokenBackedResultsPage.create(lastColumn, ImmutableList.of(),
+                                    values.size() >= batchColumnRangeSelection.getBatchHint());
```

Comment:
I think this code can be simplified. Isn't this appropriately fixed by line 873 just being a getOrDefault(row, Collections.emptyMap()); // row might not exist if all values are at a newer timestamp

Reply:
line 876 does the same thing as your new method in RowColumnRangeExtractor, after all.

Reply:
Yes, you're correct, can swap to that and remove the added code in the extractor.